multiqc/modules/base_module.py

```diff
@@ -281,14 +281,70 @@ def add_section(
             }
         )
 
-    def clean_s_name(self, s_name, f=None, root=None, filename=None, seach_pattern_key=None):
-        """Helper function to take a long file name and strip it
-        back to a clean sample name. Somewhat arbitrary.
-        :param s_name: The sample name to clean
+    @staticmethod
+    def _clean_fastq_name_pair(clean_names: List[str]) -> str:
+        """
+        Extract a sample name from a list of file names - for example, FASTQ pairs.
+        """
+        if len(set(clean_names)) == 1:  # all the same
+            return clean_names[0]
+
+        if len(clean_names) == 2:
+            # Possibly a FASTQ pair - try trimming FASTQ suffixes.
+            r1, r2 = sorted(clean_names)
+
+            # Try trimming the conventional illumina suffix with a tail 001 ending. Refs:
+            # https://support.illumina.com/help/BaseSpace_Sequence_Hub_OLH_009008_2/Source/Informatics/BS/NamingConvention_FASTQ-files-swBS.htm
+            # https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/fastq-input#:~:text=10x%20pipelines%20need%20files%20named,individual%20who%20demultiplexed%20your%20flowcell.
+            cleaned_r1 = re.sub(r"_R1_\d\d\d$", "", r1)
+            cleaned_r2 = re.sub(r"_R2_\d\d\d$", "", r2)
+            if cleaned_r1 == cleaned_r2:  # trimmed successfully
+                return cleaned_r1
+
+            # Try removing _R1 and _R2 from the middle.
+            cleaned_r1 = re.sub(r"_R1_", "_", r1)
+            cleaned_r2 = re.sub(r"_R2_", "_", r2)
+            if cleaned_r1 == cleaned_r2:  # trimmed successfully
+                return cleaned_r1
+
+            # Try trimming other variations from the end (-R1, _r1, _1, .1, etc).
+            cleaned_r1 = re.sub(r"([_.-][rR]?1)?$", "", r1)
+            cleaned_r2 = re.sub(r"([_.-][rR]?2)?$", "", r2)
+            if cleaned_r1 == cleaned_r2:  # trimmed successfully
+                return cleaned_r1
+
+        # More than 2 names, so can't assume a FASTQ pair. Just trying to keep a common prefix.
```

Comment:
Not necessarily more than 2 names, could be 2 but weren't able to trim.

```suggestion
        # Couldn't clean, or more than 2 names. Just trying to keep a common prefix.
```